src/components/public-subdomain-gateway-form/PublicSubdomainGatewayForm.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react'
+import { connect } from 'redux-bundler-react'
+import { withTranslation } from 'react-i18next'
+import Button from '../button/Button.js'
+import { checkValidHttpUrl, checkSubdomainGateway, DEFAULT_SUBDOMAIN_GATEWAY } from '../../bundles/gateway.js'
+
+const PublicSubdomainGatewayForm = ({ t, doUpdatePublicSubdomainGateway, publicSubdomainGateway }) => {
+  const [value, setValue] = useState(publicSubdomainGateway)
+  const initialIsValidGatewayUrl = !checkValidHttpUrl(value)
+  const [showFailState, setShowFailState] = useState(initialIsValidGatewayUrl)
+  const [isValidGatewayUrl, setIsValidGatewayUrl] = useState(initialIsValidGatewayUrl)
```

Comment:
can we merge these into one state variable?

```suggestion
  const [isValidGatewayUrl, setIsValidGatewayUrl] = useState(initialIsValidGatewayUrl)
```